projects/angular/src/data/tree-view/tree.ts

```diff
@@ -31,44 +32,46 @@ import { ClrTreeNode } from './tree-node';
   `,
   providers: [TREE_FEATURES_PROVIDER, TreeFocusManagerService],
   host: {
-    '[attr.tabindex]': 'tabindex',
+    tabindex: '0',
     '[attr.role]': '"tree"',
     '[attr.aria-multiselectable]': 'isMultiSelectable',
   },
 })
 export class ClrTree<T> implements AfterContentInit, OnDestroy {
+  private subscriptions: Subscription[] = [];
+
   constructor(
     public featuresService: TreeFeaturesService<T>,
     private focusManagerService: TreeFocusManagerService<T>,
-    private el: ElementRef
-  ) {}
+    { nativeElement }: ElementRef<HTMLElement>,
+    renderer: Renderer2,
+    ngZone: NgZone
+  ) {
+    const subscription = ngZone.runOutsideAngular(() =>
+      fromEvent(nativeElement, 'focusin').subscribe((event: FocusEvent) => {
+        if (event.target === nativeElement) {
+          // After discussing with the team, I've made it so that when the tree receives focus, the first visible node will be focused.
+          // This will prevent from the page scrolling abruptly to the first selected node if it exist in a deeply nested tree.
+          this.focusManagerService.focusFirstVisibleNode();
+          // when the first child gets focus,
+          // tree should no longer have tabindex of 0.
+          renderer.removeAttribute(nativeElement, 'tabindex');
+        }
+      })
+    );
 
-  private subscriptions: Subscription[] = [];
+    this.subscriptions.push(subscription);
```

Comment:
I think this should be in `ngOnInit`, not the constructor. If, for whatever reason, the constructor is called, but the component is not actually initialized, the subscription would not be unsubscribed.

Reply:
Also, I think it generally bad practice to start long-running work in constructor.

Reply:
You are right. Sadly, these optimizations are out of our current focus area. I want to try merging what is okay, but I'll have to close the ones that impose some support risk, or require more extensive testing on our side.